efi/secureboot_policy.go

```diff
@@ -257,41 +258,50 @@ func buildSignatureDbUpdateList(keystores []string) ([]*secureBootDbUpdate, erro
 	return updates, nil
 }
 
-// isSecureBootConfigMeasurementEvent determines if event corresponds to the measurement of a secure boot configuration.
-func isSecureBootConfigMeasurementEvent(event *tcglog.Event, guid efi.GUID, name string) bool {
-	if event.PCRIndex != secureBootPCR {
-		return false
-	}
-	if event.EventType != tcglog.EventTypeEFIVariableDriverConfig {
-		return false
-	}
-
-	efiVarData, isEfiVar := event.Data.(*tcglog.EFIVariableData)
-	if !isEfiVar {
-		return false
-	}
-
-	return efiVarData.VariableName == guid && efiVarData.UnicodeName == name
+func isSecureBootEvent(event *tcglog.Event) bool {
+	return event.PCRIndex == secureBootPCR
 }
 
-// isKEKMeasurementEvent determines if event corresponds to the measurement of KEK.
-func isKEKMeasurementEvent(event *tcglog.Event) bool {
-	return isSecureBootConfigMeasurementEvent(event, efi.GlobalVariable, kekName)
+// isSecureBootConfigMeasurementEvent determines if event corresponds to the measurement of secure
+// boot configuration.
+func isSecureBootConfigMeasurementEvent(event *tcglog.Event) bool {
+	return isSecureBootEvent(event) && event.EventType == tcglog.EventTypeEFIVariableDriverConfig
 }
 
-// isDbMeasurementEvent determines if event corresponds to the measurement of UEFI authorized signature database.
+// isDbMeasurementEvent determines if event corresponds to the measurement of the UEFI authorized
+// signature database.
 func isDbMeasurementEvent(event *tcglog.Event) bool {
-	return isSecureBootConfigMeasurementEvent(event, efi.ImageSecurityDatabaseGuid, dbName)
+	if !isSecureBootConfigMeasurementEvent(event) {
+		return false
+	}
+
+	data := event.Data.(*tcglog.EFIVariableData)
+	return data.VariableName == efi.ImageSecurityDatabaseGuid && data.UnicodeName == dbName
 }
 
-// isDbxMeasurementEvent determines if event corresponds to the measurement of UEFI forbidden signature database.
-func isDbxMeasurementEvent(event *tcglog.Event) bool {
-	return isSecureBootConfigMeasurementEvent(event, efi.ImageSecurityDatabaseGuid, dbxName)
+// isSignatureDatabaseMeasurementEVent determines if event corresponds to the measurement of one
+// of the UEFI signature databases.
+func isSignatureDatabaseMeasurementEvent(event *tcglog.Event) bool {
+	if !isSecureBootConfigMeasurementEvent(event) {
+		return false
+	}
+
+	data := event.Data.(*tcglog.EFIVariableData)
+	switch {
+	case data.VariableName == efi.GlobalVariable && data.UnicodeName == pkName:
+		return true
+	case data.VariableName == efi.GlobalVariable && data.UnicodeName == kekName:
+		return true
+	case data.VariableName == efi.ImageSecurityDatabaseGuid:
```

Comment:
this covers db and dbx, but db is treated separately? are there more cases that will hit here now?

Reply:
Not in this PR, but there will be in a future PR (see https://github.com/chrisccoulson/secboot/commit/df2bc9d598e821f828e9d536bb1f1168c14976ce)